sky/shell/tracing_controller.cc

```diff
@@ -33,7 +108,22 @@ void TracingController::StartTracing() {
   StartBaseTracing();
 }
 
-void TracingController::StopTracing(const base::FilePath& path) {
+void TracingController::StopTracing(const base::FilePath& path,
+                                    bool terminateLoopWhenDone) {
+  auto runner = base::MessageLoop::current()->task_runner();
+  runner->PostTask(FROM_HERE, base::Bind(&TracingController::StopTracingAsync,
+                                         weak_factory_.GetWeakPtr(), path,
+                                         terminateLoopWhenDone));
+}
+
+void TracingController::StopTracingAsync(const base::FilePath& path,
+                                    bool terminateLoopWhenDone) {
+  if (terminate_loop_on_write_) {
```

Comment:
Could happen if observatory is attempting to stop tracing and another request to stop tracing is made from C++ (the old way of capturing traces to the documents directory from C++ still works)